modules/circe/src/main/scala/tethys/circe/ast/CirceSupport.scala

```diff
@@ -0,0 +1,76 @@
+package tethys.circe.ast
+
+import io.circe.{Json, JsonNumber, JsonObject}
+import io.circe.syntax._
+import tethys.readers.{FieldName, ReaderError}
+import tethys.readers.tokens.TokenIterator
+import tethys.{JsonObjectWriter, JsonReader, JsonWriter}
+import tethys.writers.tokens.TokenWriter
+
+trait CirceSupport {
+  implicit lazy val circeJsonObjectWriter: JsonWriter[JsonObject] = new JsonObjectWriter[JsonObject] {
+    def writeValues(value: JsonObject, writer: TokenWriter): Unit =
+      value.toIterable.foreach { case (k, v) => circeJsonWriter.write(k, v, writer) }
```

Comment:
use your **JsonFolder** implementation direct instead of call `circeJsonWriter.write` to avoid allocations of **JsonFolder** instances on each value of Object